Sources/MapboxNavigation/PuckType.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+import MapboxMaps
+
+public enum PuckType {
+    
+    case puck2D(configuration: Puck2DConfiguration)
+    
+    case puck3D(configuration: Puck3DConfiguration)
+    
+    case `default`
```

Comment:
Currently, the `NavigationMapView.userCourseView` property sits alongside `NavigationMapView.puckType` but only applies when `puckType` is set to `default`. Ideally, we should avoid parallel state, because it often gets out of sync and confuses developers. Consider moving `userCourseView` to an associated value of this case. It would be unusual for a `default` case to have an associated value, so we could rename it to `courseView` at the same time.

Reply:
Thanks for the suggestion, I've add the sync from the `puckType` to the `userCourseView` and add configuration for it to customize.